Source/Core/VideoCommon/BPFunctions.cpp

```diff
@@ -39,19 +39,28 @@ void SetGenerationMode()
 
 void SetScissor()
 {
-  /* NOTE: the minimum value here for the scissor rect and offset is -342.
-   * GX internally adds on an offset of 342 to both the offset and scissor
-   * coords to ensure that the register was always unsigned.
+  /* NOTE: the minimum value here for the scissor rect is -342.
+   * GX internally adds on an offset of 342 to scissor coords to
+   * ensure that the register was always unsigned.
    *
    * The code that was here before tried to "undo" this offset, but
    * since we always take the difference, the +342 added to both
    * sides cancels out. */
 
-  /* The scissor offset is always even, so to save space, the scissor offset
+  /* NOTE: With positive scissor offset the scissor rectangle is shifted left and/or up;
+   * and with negative scissor offset, the scissor rectangle is shifted right and/or down.
+   *
+   * GX internally also adds on an offset of 342 to scissor offset.
+   * The scissor offset is always even, so to save space, the scissor offset
    * register is scaled down by 2. So, if somebody calls
-   * GX_SetScissorBoxOffset(20, 20); the registers will be set to 10, 10. */
-  const int xoff = bpmem.scissorOffset.x * 2;
-  const int yoff = bpmem.scissorOffset.y * 2;
+   * GX_SetScissorBoxOffset(20, 20); the registers will be set to (10 + 171, 10 + 171).
+   *
+   * The scissor offset register is 10bit signed [-512, 511].
+   * (e.g. In Super Mario Galaxy 1 and 2, during the Boss roar effect,
+   * the scissor offset register will be set to -61, so the scissor offset is -464)
+   */
```

Comment:
Any suggestion about this comments is welcome, so we can make this comments more clear.

Reply:
"With positive scissor offset" should be "With a positive scissor offset" (and there should be a comma after offset on both lines).

It might be worth clarifying that "GX internally also adds on an offset of 342 to scissor offset." means "The GX SDK functions internally add an offset of 342".

It would also be useful to elaborate on the math, e.g. "the scissor offset register will be set to -61, so the scissor offset is `(-61)*2 - 342 = -122 - 342 = -464`", or maybe it makes more sense in the opposite direction: "for a scissor offset of -464, the scissor offset register will be set to `(-464 + 342)/2 = -122/2 = -61`".